ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -806,6 +811,16 @@ public static boolean isChunkUpload(Map<String, Object> args) throws RestService
     return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false);
   }
 
+  /**
+   * Determine if {@link Headers#NAMED_UPSERT} is set in the request args.
+   * @param args The request arguments.
+   * @return {@code false} if {@link Headers#NAMED_UPSERT} is set to false else return true.
+   * @throws RestServiceException if exception occurs during parsing the arg.
+   */
+  public static boolean isUpsertForNamedBlob(final Map<String, Object> args) throws RestServiceException {
+    return getOptionalBooleanHeader(args, Headers.NAMED_UPSERT, true);
+  }
```

Comment:
The PR looks much better now that i can focus on your changes and not be distracted by all the other other irrelevant code.